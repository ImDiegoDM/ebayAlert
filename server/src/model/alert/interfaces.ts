import * as validator from 'validator'
import { ObjError } from '../../objError';

export interface SearchPhrase{
  phrase:string;
  howOften:2|10|30
}

export interface Alert{
  id?:string,
  email:string,
  phrases:SearchPhrase[]
}

export const collection = 'search_phrases';
