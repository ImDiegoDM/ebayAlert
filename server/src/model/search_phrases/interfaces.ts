export interface ISearchPhrase {
  email: string;
  phrase: string;
  howOften: 2 | 10 | 30;
}

export const collection = 'search_phrases';
