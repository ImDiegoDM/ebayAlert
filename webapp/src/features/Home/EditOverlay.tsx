import axios from 'axios';
import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { fetchPhrases } from '../../actions';
import { ISearchPhrase } from '../../interfaces';
import { Overlay } from './components/Overlay';
import { SearchPhraseForm } from './components/SearchPhraseForm';
import { ErrorText } from './ErrorText';

interface EditOverlayProps {
  open: boolean;
  onClose?: () => void;
  phrase?: ISearchPhrase;
}

function EditSearchPhrase(value: ISearchPhrase) {
  return axios.put(process.env.REACT_APP_SERVER_URL + '/search_phrases/' + value._id, {
    email: value.email,
    howOften: value.howOften,
    phrase: value.phrase,
  });
}

export function EditOverlay(props: EditOverlayProps) {
  const [value, setValue] = useState<ISearchPhrase|undefined>(undefined);
  const [error, setError] = useState('');

  useEffect(() => {
    setValue(props.phrase);
  }, [props.phrase]);

  const dispatch = useDispatch();

  function handleSubmit() {
    if (value === undefined) {
      return;
    }
    EditSearchPhrase(value).then((res) => {
      if (props.onClose) {
        props.onClose();
        setError('');
      }

      dispatch(fetchPhrases());
    }).catch((err) => {
      setError(err.response.data);
    });
  }

  return <Overlay open={props.open} onExit={() => {
    if (props.onClose) {
      props.onClose();
    }
    setError('');
  }}>
    Edit a Search Phrase
    {error && <ErrorText>{error}</ErrorText>}
    {value !== undefined && <SearchPhraseForm value={value} onValueChange={setValue} onSubmit={handleSubmit}/>}
  </Overlay>;
}
