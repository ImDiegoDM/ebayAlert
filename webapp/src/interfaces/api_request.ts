import { AxiosError } from 'axios';

export interface IApiRequest<T= any, E= any> {
  fetching: boolean;
  fetched: boolean;
  ok: boolean;
  data?: T;
  error?: AxiosError;
}
