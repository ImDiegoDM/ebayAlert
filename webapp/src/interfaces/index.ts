export * from './api_request';
export * from './search_phrase';
