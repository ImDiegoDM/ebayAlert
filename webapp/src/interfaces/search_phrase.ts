export interface ISearchPhrase {
  phrase: string;
  email: string;
  howOften: '2' | '10' | '30';
}
