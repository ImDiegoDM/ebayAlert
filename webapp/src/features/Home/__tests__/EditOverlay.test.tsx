import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { applyMiddleware, createStore } from 'redux';
import promise from 'redux-promise-middleware';
import { ISearchPhrase } from '../../../interfaces';
import reducer from '../../../reducers';
import { EditOverlay } from '../EditOverlay';

const phrase: ISearchPhrase = {
  _id: 'sdfd5fa1dfad',
  email: 'test@test.com',
  howOften: '2',
  phrase: 'drone',
};

describe('<EditOverlay/>', () => {
  it('should match snapshot when open', () => {

    const middlware = applyMiddleware(promise);

    const store = createStore(reducer, middlware);

    const { container } = render(
      <Provider store={store}>
        <EditOverlay phrase={phrase} open={true}/>
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot when closed', () => {

    const middlware = applyMiddleware(promise);

    const store = createStore(reducer, middlware);

    const { container } = render(
      <Provider store={store}>
        <EditOverlay phrase={phrase} open={false}/>
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
