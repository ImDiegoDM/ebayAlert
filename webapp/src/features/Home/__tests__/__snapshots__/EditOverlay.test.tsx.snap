// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditOverlay/> should match snapshot when closed 1`] = `
.c0 {
  width: 300px;
  background-color: white;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transform-origin: right;
  -ms-transform-origin: right;
  transform-origin: right;
  box-shadow: 0 0 11px 4px #00000047;
  padding: 20px;
  z-index: 15;
}

.c1 {
  position: absolute;
  top: 20px;
  right: 20px;
  cursor: pointer;
  color: gray;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 15px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-times fa-w-11 "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </div>
  Edit a Search Phrase
  
  <form
    class="c2"
  >
    <label
      class="c3"
    >
      Email
      <input
        required=""
        type="email"
        value="test@test.com"
      />
    </label>
    <label
      class="c3"
    >
      Phrase
      <input
        required=""
        type="text"
        value="drone"
      />
    </label>
    <label
      class="c3"
    >
      How often
      <select>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="30"
        >
          30
        </option>
      </select>
    </label>
    <div>
      <input
        type="submit"
      />
    </div>
  </form>
</div>
`;

exports[`<EditOverlay/> should match snapshot when open 1`] = `
.c0 {
  width: 300px;
  background-color: white;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
  -webkit-transform-origin: right;
  -ms-transform-origin: right;
  transform-origin: right;
  box-shadow: 0 0 11px 4px #00000047;
  padding: 20px;
  z-index: 15;
}

.c1 {
  position: absolute;
  top: 20px;
  right: 20px;
  cursor: pointer;
  color: gray;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 15px;
}

<div
  class="c0"
  open=""
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-times fa-w-11 "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
      />
    </svg>
  </div>
  Edit a Search Phrase
  
  <form
    class="c2"
  >
    <label
      class="c3"
    >
      Email
      <input
        required=""
        type="email"
        value="test@test.com"
      />
    </label>
    <label
      class="c3"
    >
      Phrase
      <input
        required=""
        type="text"
        value="drone"
      />
    </label>
    <label
      class="c3"
    >
      How often
      <select>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="30"
        >
          30
        </option>
      </select>
    </label>
    <div>
      <input
        type="submit"
      />
    </div>
  </form>
</div>
`;
